src/meta/src/barrier/checkpoint/control.rs

```diff
@@ -441,16 +435,27 @@ impl CheckpointControl {
 
     pub(crate) fn inflight_infos(
         &self,
-    ) -> impl Iterator<Item = (DatabaseId, &InflightSubscriptionInfo, &InflightDatabaseInfo)> + '_
-    {
+    ) -> impl Iterator<
+        Item = (
+            DatabaseId,
+            &InflightSubscriptionInfo,
+            &InflightDatabaseInfo,
+            impl Iterator<Item = &InflightStreamingJobInfo> + '_,
```

Comment:
This is a fix for snapshot backfill inflight info, separate from database info. So we didn't recover it previously.